hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/DynamicConfigurationAwareConfig.java

```diff
@@ -123,6 +124,8 @@ public MapConfig getStaticConfig(@Nonnull Config staticConfig, @Nonnull String n
 
     public DynamicConfigurationAwareConfig(Config staticConfig, HazelcastProperties properties) {
         assert !(staticConfig instanceof DynamicConfigurationAwareConfig) : "A static Config object is required";
+        merge(staticConfig.getHotRestartPersistenceConfig(), staticConfig.getPersistenceConfig());
```

Comment:
Thanks for the fix. Can you add a test to ensure that the `config.getHotRestart/PersistenceConfig().setXYZ` style config results in properly merged configs when obtained from `node.getConfig()` on a running member?

Reply:
thanks, yes, added a test for it on the EE side